modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/restapi/admin/throttlingpolicy/AdvancedThrottlingPolicyTestCase.java

```diff
@@ -0,0 +1,308 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.wso2.am.integration.tests.restapi.admin.throttlingpolicy;
+
+import org.apache.http.HttpStatus;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import org.wso2.am.integration.clients.admin.ApiException;
+import org.wso2.am.integration.clients.admin.ApiResponse;
+import org.wso2.am.integration.clients.admin.api.dto.AdvancedThrottlePolicyDTO;
+import org.wso2.am.integration.clients.admin.api.dto.BandwidthLimitDTO;
+import org.wso2.am.integration.clients.admin.api.dto.ConditionalGroupDTO;
+import org.wso2.am.integration.clients.admin.api.dto.HeaderConditionDTO;
+import org.wso2.am.integration.clients.admin.api.dto.IPConditionDTO;
+import org.wso2.am.integration.clients.admin.api.dto.JWTClaimsConditionDTO;
+import org.wso2.am.integration.clients.admin.api.dto.QueryParameterConditionDTO;
+import org.wso2.am.integration.clients.admin.api.dto.RequestCountLimitDTO;
+import org.wso2.am.integration.clients.admin.api.dto.ThrottleConditionDTO;
+import org.wso2.am.integration.clients.admin.api.dto.ThrottleLimitDTO;
+import org.wso2.am.integration.test.impl.AdminApiTestHelper;
+import org.wso2.am.integration.test.impl.DtoFactory;
+import org.wso2.am.integration.test.utils.base.APIMIntegrationBaseTest;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+public class AdvancedThrottlingPolicyTestCase extends APIMIntegrationBaseTest {
+
+    private String displayName = "Test Policy";
+    private String description = "This is a test advanced throttle policy";
+    private String timeUnit = "min";
+    private Integer unitTime = 1;
+    private AdvancedThrottlePolicyDTO requestCountPolicyDTO;
+    private AdvancedThrottlePolicyDTO bandwidthPolicyDTO;
+    private AdvancedThrottlePolicyDTO conditionalGroupsPolicyDTO;
+    private AdminApiTestHelper adminApiTestHelper;
+
+    @Factory(dataProvider = "userModeDataProvider")
+    public AdvancedThrottlingPolicyTestCase(TestUserMode userMode) {
+
+        this.userMode = userMode;
+    }
+
+    @DataProvider
+    public static Object[][] userModeDataProvider() {
+
```

Comment:
Remove the blank lines

Reply:
Fixed with https://github.com/wso2/product-apim/pull/8750/commits/6d5d3d36ba4acc833b947ad9f83e3b19e3ad93c9